Debug App/Sources/View Controllers/New UI/MerchantHeadlessCheckoutKlarnaViewController.swift

```diff
@@ -0,0 +1,605 @@
+        showAlert(
+            title: "Error",
+            message: error.errorDescription ?? error.localizedDescription
+        )
+    }
+}
+
+// MARK: - PrimerHeadlessValidatableDelegate
+extension MerchantHeadlessCheckoutKlarnaViewController: PrimerHeadlessValidatableDelegate {
+    func didUpdate(validationStatus: PrimerSDK.PrimerValidationStatus, for data: PrimerSDK.PrimerCollectableData?) {
+        switch validationStatus {
+        case .invalid(let errors):
+            var message = ""
+            for error in errors {
+                message += (error.errorDescription ?? error.localizedDescription) + "\n"
+            }
+            self.showAlert(title: "Validation Error", message: "\(message)")
+            
+        case .valid:
+            if let _ = data as? KlarnaPaymentSessionCollectableData {
+                self.startPaymentSession()
+            }
+            
+        default:
+            break
+        }
+    }
+}
+
+// MARK: - PrimerHeadlessSteppableDelegate
+extension MerchantHeadlessCheckoutKlarnaViewController: PrimerHeadlessSteppableDelegate {
+    func didReceiveStep(step: PrimerSDK.PrimerHeadlessStep) {
+        if let step = step as? KlarnaPaymentSessionCreation {
+            switch step {
+            case .paymentSessionCreated(let clientToken, let paymentCategories):
+                DispatchQueue.main.async { [weak self] in
+                    self?.hideLoader()
+                    self?.clientToken = clientToken
+                    self?.paymentCategories = paymentCategories
+                }
+            }
+        }
+        
+        if let step = step as? KlarnaPaymentViewHandling {
+            switch step {
+            case .viewInitialized:
+                klaraViewHandlingComponent.loadPaymentView()
+                
+            case .viewResized(let height):
+                paymentViewContainerHeightConstraint.constant = height
+                view.layoutIfNeeded()
+                
+            case .viewLoaded:
+                hideLoader()
+                paymentContainerView.isHidden = false
+                view.bringSubviewToFront(paymentContainerView)
+                
+            default:
+                break
+            }
+        }
+        
+        if let step = step as? KlarnaPaymentSessionAuthorization {
+            hideLoader()
+            
+            switch step {
+            case .paymentSessionAuthorized(let authToken):
+                showAlert(title: "Success", message: "Payment session completed with token: \(authToken)") { [unowned self] in
+                    navigationController?.popToRootViewController(animated: true)
+                }
+                
+            case .paymentSessionAuthorizationFailed:
+                showAlert(title: "Authorization", message: "Payment authorization failed") { [unowned self] in
+                    klarnaSessionAuthorizationComponent.reauthorizeSession()
```

Comment:
I dont think we can call reauthorize here, According to the klarna docs, we should use reauthorize when changing an already authorized session, it doesnt work to create a new one.